### PR TITLE
json shredder

### PR DESCRIPTION
**Pull request by AudreyAltman**
This adds a JSON shredder, which functions like the existing XML shredder.

This has been tested locally with a random sample of ~1,000 MDL records.  It has been indexed here: http://ec2-52-87-236-197.compute-1.amazonaws.com:9200/json-shredder-test/_search?pretty=true&q=*:*

In order to get elastic search to index the records, I had to turn off the date serialization (some fields had both date and string values, and this was causing errors b/c elastics search expected all fields values to be of the same type).  Here is the invocation I used:

```
curl -XPUT 'http://ec2-52-87-236-197.compute-1.amazonaws.com:9200/json-shredder-test/' -d '{"mappings":{"_default_":{"date_detection":false}}}'
```

I made some small changes to the XML shredder for consistency, notably concatenating field names with `_` instead of `.` (the latter was causing a perplexing serialization bugs when it hit elastic search).

This addresses [ticket DT-1849](https://digitalpubliclibraryofamerica.atlassian.net/browse/DT-1489).

### CODE REVIEW

File: src/main/scala/dpla/ingestion3/preMappingReports/XmlShredder.scala [hunk 1]
```diff
@@ -168,13 +168,5 @@ object XmlShredder {
   def labelWithPrefix(node: Node): String =
     // We have to use null here b/c it can be returned by scala.xml.Node
     if(node.prefix == null) node.label
-    else s"${node.prefix}_${node.label}"
```
Comment (moltude) [inline src/main/scala/dpla/ingestion3/preMappingReports/XmlShredder.scala]: I'm probably missing something glaring here but why was this changed from `_` to `.`.? Everything else seems to go the other way...

Comment (AudreyAltman) [reply src/main/scala/dpla/ingestion3/preMappingReports/XmlShredder.scala]: Yeah, this concatenating a prefix with its label.  All the others are concatenating a parent label with a child label.  So there needs to be a different character here.

File: src/main/scala/dpla/ingestion3/preMappingReports/XmlShredder.scala [hunk 2]
```diff
@@ -20,17 +20,17 @@ object XmlShredder {
     * Examples:
```
Comment (mdellabitta) [inline src/main/scala/dpla/ingestion3/preMappingReports/XmlShredder.scala]: Was there a find/replace in here that went awry? I thought we were using . as a path separator and _ for separating xml element names from namespace nicknames.

Comment (AudreyAltman) [reply src/main/scala/dpla/ingestion3/preMappingReports/XmlShredder.scala]: Yeah, I got it mixed up when I wrote the JSON shredder, and then couldn't recall which one was the original.  It's consistent now with both shredders.  Do you have strong feelings about which is path separator and which is element separator?

Comment (mdellabitta) [reply src/main/scala/dpla/ingestion3/preMappingReports/XmlShredder.scala]: Nope. Let's let Gretchen tell us.
